examples/wave-eager-mpi.py

```diff
@@ -126,22 +126,50 @@ def rhs(t, w):
     t = 0
     t_final = 3
     istep = 0
-    while t < t_final:
-        fields = rk4_step(fields, t, dt, rhs)
 
-        if istep % 10 == 0:
-            print(istep, t, discr.norm(fields[0]))
-            vis.write_vtk_file("fld-wave-eager-mpi-%03d-%04d.vtu" % (rank, istep),
+    # initialize Leap integrator if using one
+    import importlib
+    leap_spec = importlib.util.find_spec("leap")
+    not_found = leap_spec is None
+    if not_found:
+        raise ValueError("Leap uninstalled")
+    if use_leap:
+        from leap.rk import RK4MethodBuilder
+        from utils import leap_setup
+        interp = leap_setup(RK4MethodBuilder, "w", rhs, t, dt, fields)
+
+    while t < t_final:
+        if use_leap:
+            for event in interp.run(t_end=t+dt):
+                if isinstance(event, interp.StateComputed):
+                    fields = event.state_component
+                    if istep % 10 == 0:
+                        print(istep, t, discr.norm(fields[0]))
+                        vis.write_vtk_file(
+                            "fld-wave-eager-mpi-%03d-%04d.vtu" % (rank, istep),
+                            [
+                                ("u", fields[0]),
+                                ("v", fields[1:]),
+                                ])
+                    t += dt
+                    istep += 1
+        else:
+            fields = rk4_step(fields, t, dt, rhs)
+
+            if istep % 10 == 0:
+                print(istep, t, discr.norm(fields[0]))
+                vis.write_vtk_file(
+                    "fld-wave-eager-mpi-%03d-%04d.vtu" % (rank, istep),
```

Comment:
To me, this is a step back in readability. Examples should focus on simplicity. This change does the opposite. (I.e. I think this should be reverted.)

Reply:
Reverted to what? Not using Leap?